cmd/encryption-v1.go

```diff
@@ -141,55 +149,79 @@ const (
 	ServerSideEncryptionSealedKey = ReservedMetadataPrefix + "Server-Side-Encryption-Sealed-Key"
 )
 
-// SSESealAlgorithmDareSha256 specifies DARE as authenticated en/decryption scheme and SHA256 as cryptographic
-// hash function.
-const SSESealAlgorithmDareSha256 = "DARE-SHA256"
+const (
+	// SSEIVSize is the size of the randomly chosen IV used to derive the key-encryption-key.
+	SSEIVSize = 32
+
+	// DAREPayloadSize is the max. size of the package payload for a DARE 2.0 package.
+	DAREPayloadSize = 1 << 16
+
+	// DAREOverhead is the overhead (header + tag) added by encrypting one payload using DARE 2.0.
+	DAREOverhead = 32
+
+	// SSESealAlgorithmDareSha256 specifies DARE as authenticated en/decryption scheme and SHA256 as
+	// cryptographic hash function.
+	SSESealAlgorithmDareSha256 = "DARE-SHA256"
+)
+
+var (
+	// SSECustomerHeaders can parse regular SSE-C requests.
+	SSECustomerHeaders = sseCustomerHeaders{
+		algorithm: SSECustomerAlgorithm,
+		key:       SSECustomerKey,
+		keyMD5:    SSECustomerKeyMD5,
+	}
+
+	// SSECopyCustomerHeaders can parse SSE-C copy requests.
+	SSECopyCustomerHeaders = sseCustomerHeaders{
+		algorithm: SSECopyCustomerAlgorithm,
+		key:       SSECopyCustomerKey,
+		keyMD5:    SSECopyCustomerKeyMD5,
+	}
+)
 
-// hasSSECustomerHeader returns true if the given HTTP header
-// contains server-side-encryption with customer provided key fields.
-func hasSSECustomerHeader(header http.Header) bool {
-	return header.Get(SSECustomerAlgorithm) != "" || header.Get(SSECustomerKey) != "" || header.Get(SSECustomerKeyMD5) != ""
+// sseCustomerHeaders contains the SSE-C headers
+// which must be present in the client request.
+type sseCustomerHeaders struct {
+	algorithm string
+	key       string
+	keyMD5    string
 }
 
-// hasSSECopyCustomerHeader returns true if the given HTTP header
-// contains copy source server-side-encryption with customer provided key fields.
-func hasSSECopyCustomerHeader(header http.Header) bool {
-	return header.Get(SSECopyCustomerAlgorithm) != "" || header.Get(SSECopyCustomerKey) != "" || header.Get(SSECopyCustomerKeyMD5) != ""
+// Contains returns true if the provided headers contain
+// the at least one SSE-C header field.
+func (sse sseCustomerHeaders) Contains(header http.Header) bool {
```

Comment:
What about 'Test()' instead of 'Contains()' ?

Reply:
Okay now we moved from `has` over `contains` to `test` xD
The problem with test is that it does not read very well IMO: `SSECustomerHeaders.Test(headers)` what means `...Test(...)`?  - I thought also about `IsPresent`. `Match` seemed not appropriate, but I've no strong opinion about it in general.
The only thing I'm aiming for is that if you read it in the code it should *obvious* what its purpose is.